add doc for broadcast join (#4068)
pingcap/docs: tiflash/use-tiflash.md

>
> Before v4.0.2, TiDB does not support the new framework for collations, so in those previous versions, if you enable the [new framework for collations](/character-set-and-collation.md#new-framework-for-collations), none of the expressions can be pushed down. This restriction is removed in v4.0.2 and later versions.

> TiFlash mainly supports predicate, aggregate push-down calculations and table joins. Push-down calculations can help TiDB perform distributed acceleration. Currently, `Full Outer Join` and `DISTINCT COUNT` are not the supported calculation types, which will be optimized in later versions.

Reviewer: ```suggestion
TiFlash supports predicate, aggregate push-down calculations, and table joins. Push-down calculations can help TiDB perform distributed acceleration. Currently, `Full Outer Join` and `DISTINCT COUNT` are not the supported calculation types, which will be optimized in later versions.
```

> You can enable the push-down connection (`Join`) function via the following Session Variables. Currently, `Full Outer Join` is not supported.

Reviewer: ```suggestion
You can enable the push-down of `join` using the following session variable (`Full Outer Join` is currently not supported):
```
